src/main/java/com/gluonhq/richtextarea/RichListCell.java

```diff
@@ -116,14 +113,41 @@ protected void updateItem(Paragraph item, boolean empty) {
             var fragments = new ArrayList<Node>();
             var backgroundIndexRanges = new ArrayList<IndexRangeColor>();
             var length = new AtomicInteger();
+            var positions = new ArrayList<Integer>();
+            positions.add(item.getStart());
+            AtomicInteger tp = new AtomicInteger(item.getStart());
             richTextAreaSkin.getViewModel().walkFragments((text, decoration) -> {
                 if (decoration instanceof TextDecoration && !text.isEmpty()) {
-                    final Text textNode = buildText(text.replace("\n", TextBuffer.ZERO_WIDTH_TEXT), (TextDecoration) decoration);
-                    fragments.add(textNode);
-                    Color background = ((TextDecoration) decoration).getBackground();
-                    if (background != Color.TRANSPARENT) {
-                        backgroundIndexRanges.add(new IndexRangeColor(
-                                length.get(), length.get() + text.length(), background));
+                    if (item.getDecoration().hasTableDecoration()) {
+                        AtomicInteger s = new AtomicInteger();
+                        IntStream.iterate(text.indexOf(TextBuffer.ZERO_WIDTH_TABLE_SEPARATOR),
+                                        index -> index >= 0,
+                                        index -> text.indexOf(TextBuffer.ZERO_WIDTH_TABLE_SEPARATOR, index + 1))
+                                .boxed()
+                                .forEach(i -> {
+                                    String tableText = text.substring(s.getAndSet(i + 1), i + 1);
```

Comment:
We need to have _one_ character per cell, otherwise the caret can't be moved to that cell, so that's why I keep this non-visible character (otherwise I'd have used `split`).